engine/src/flutter/impeller/entity/contents/text_contents.cc

```diff
@@ -298,26 +296,45 @@ bool TextContents::Render(const ContentContext& renderer,
           sampler_desc)  // sampler
   );
 
-  auto& host_buffer = renderer.GetTransientsBuffer();
-  size_t vertex_count = 0;
+  HostBuffer& host_buffer = renderer.GetTransientsBuffer();
+  size_t glyph_count = 0;
   for (const auto& run : frame_->GetRuns()) {
-    vertex_count += run.GetGlyphPositions().size();
+    glyph_count += run.GetGlyphPositions().size();
   }
-  vertex_count *= 6;
+  size_t vertex_count = glyph_count * 4;
+  size_t index_count = glyph_count * 6;
 
   BufferView buffer_view = host_buffer.Emplace(
       vertex_count * sizeof(VS::PerVertexData), alignof(VS::PerVertexData),
-      [&](uint8_t* contents) {
+      [&](uint8_t* data) {
         VS::PerVertexData* vtx_contents =
-            reinterpret_cast<VS::PerVertexData*>(contents);
-        ComputeVertexData(vtx_contents, frame_, scale_,
-                          /*entity_transform=*/entity_transform, offset_,
-                          GetGlyphProperties(), atlas);
+            reinterpret_cast<VS::PerVertexData*>(data);
+        ComputeVertexData(/*vtx_contents=*/vtx_contents,
+                          /*frame=*/frame_,
+                          /*scale=*/scale_,
+                          /*entity_transform=*/entity_transform,
+                          /*offset=*/offset_,
+                          /*glyph_properties=*/GetGlyphProperties(),
+                          /*atlas=*/atlas);
+      });
+  BufferView index_buffer_view = host_buffer.Emplace(
+      index_count * sizeof(uint16_t), alignof(uint16_t), [&](uint8_t* data) {
+        uint16_t* indices = reinterpret_cast<uint16_t*>(data);
+        size_t j = 0;
+        for (auto i = 0u; i < glyph_count; i++) {
+          size_t base = i * 4;
+          indices[j++] = base + 0;
+          indices[j++] = base + 1;
+          indices[j++] = base + 2;
+          indices[j++] = base + 1;
+          indices[j++] = base + 2;
+          indices[j++] = base + 3;
```

Comment:
nit: Switching to triangle strip would be more efficient.

Reply:
The glyph rects aren't connected, so switching to strip would be move overall geometry:

Before: 6 pts
This PR: 4 pts
w/ Strip: 4pts + 4 to break primitive